File: Problems/0258. (E) Add Digits/Java/solution.java
class Solution {
  public int addDigits(int num) {
    // iteration

    int ret = 0;
    while (true) {
      ret += num % 10;
      num /= 10;
      if (num == 0) {
        if (ret < 10)
          break;
        num = ret;
        ret = 0;
      }
    }
    return ret;
  }
}

File: Problems/0258. (E) Add Digits/Python3/solution.py
class Solution:
  def addDigits(self, num: int) -> int:
    # iteration

    ret = 0
    while True:
      ret += num % 10
      num //= 10
      if not num:
        if ret < 10:
          break
        num = ret
        ret = 0
    
    return ret

